### PR TITLE
I 12947 workaround

### PR DESCRIPTION
**Pull request by cameroncaci**
https://github.com/transcom/react-file-viewer/pull/3 was indirectly reverted due to this repository not being compatible with `pdsjf-dist` versions `^4.2.67`, where the fix to a [critical vulnerability](https://security.snyk.io/vuln/SNYK-JS-PDFJSDIST-6810403) identified from [the mymove repo](https://github.com/transcom/mymove/security/dependabot/146), for now the best course of action is to implement the workaround to avoid this vulnerability. The issue ticket will have the analysis attached and proposed method forward, however for now it is best to roll-back this repository's releases to an old version, `v1.2.1`, and implement the workaround increasing the version to `v1.2.2`. `v1.2.2` will become the latest release. Every version current after `v1.2.2` is not functional, even though it builds.

# Changes proposed in this pull request:

- Rollback the `pdfjs-dist` library update
- Implement vulnerability workaround
- Cherry-pick the tool versions commits from https://github.com/transcom/react-file-viewer/pull/3

# To simulate how it will function against mymove, conduct the following
- Clone this repository and pull this branch
- Run `yarn link` to create a symbolic dependency link
- Cd into your local mymove repository, run `yarn link @transcom/react-file-viewer`, then reinstall via `yarn install`
- Run your tests, as the "react-file-viewer" dependency will now be running this branch locally

# How to test
Run `yarn test` inside this branch

### CODE REVIEW

**Comment by deandreJones** [comment]
conflicts

**Comment by cameroncaci** [comment]
Added eslint bypass during merge

**Comment by deandreJones** [comment]
> ## [I-12947](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A962159)
> This is an indirect revert of efforts contributed to #3. Due to this repository not being compatible with `pdsjf-dist` versions `^4.2.67`, where the fix to a [critical vulnerability](https://security.snyk.io/vuln/SNYK-JS-PDFJSDIST-6810403) identified from [the mymove repo](https://github.com/transcom/mymove/security/dependabot/146), for now the best course of action is to implement the workaround to avoid this vulnerability. The issue ticket will have the analysis attached and proposed method forward, however for now it is best to roll-back this repository's releases to an old version, `v1.2.1`, and implement the workaround increasing the version to `v1.2.2`. `v1.2.2` will become the latest release. Every version current after `v1.2.2` is not functional, even though it builds.
> 
> # Changes proposed in this pull request:
> * Rollback the `pdfjs-dist` library update
> * Implement vulnerability workaround
> * Cherry-pick the tool versions commits from [I-12947 node canvas #3](https://github.com/transcom/react-file-viewer/pull/3)
> 
> # To simulate how it will function against mymove, conduct the following
> * Clone this repository and pull this branch
> * Run `yarn link` to create a symbolic dependency link
> * Cd into your local mymove repository, run `yarn link @trussworks/react-file-viewer`, then reinstall via `yarn install`
> * Run your tests, as the "react-file-viewer" dependency will now be running this branch locally
> 
> # How to test
> Run `yarn test` inside this branch

yarn link to @trussworks?

File: package.json
Comment (deandreJones) [inline package.json:1]: are we not reverting to the prev pdfjs-dist 1.8.357?

**Comment by cameroncaci** [comment]
> yarn link to @trussworks?

Whoops, good catch thank you. I had copied it from my terminal, `1.2.1` was still using "trussworks" instead of "transcom", hence why my cli spit it out

File: package.json
Comment (cameroncaci) [reply package.json:1]: The merge with main switched it back. I was hoping I could just push the new commit without having to revert the old ones. Will fix, one moment

Comment (cameroncaci) [reply package.json:1]: Utilizing `git revert` rather than force pushing in order to preserve git history

**Comment by cameroncaci** [comment]
Reverted https://github.com/transcom/react-file-viewer/commit/85abde8ea13c87973e4e745ee9f898ad11656e90 with some minor modifications

File: src/components/drivers/pdf-viewer.jsx [hunk 1]
```diff
@@ -103,17 +107,20 @@ export default class PDFDriver extends React.Component {
   }
 
   componentDidMount() {
-    const { filePath } = this.props
-    const containerWidth = this.container.offsetWidth
-
-    const loadingTask = PDFJS.getDocument(filePath)
-    loadingTask.onProgress = (progressData) => {
-      this.progressCallback(progressData)
+    // Only utilized PDFJS.getDocument() if isEvalSupported == false.
```
Comment (traskowskycaci) [inline src/components/drivers/pdf-viewer.jsx:110]: utilized here should be utilize, right?

Comment (cameroncaci) [reply src/components/drivers/pdf-viewer.jsx:110]: Yes, yes it should. Thanks for spotting that

**Review by traskowskycaci — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]
high hopes!

**Pull request merged**